lib/livebook/settings.ex

```diff
@@ -242,4 +246,39 @@ defmodule Livebook.Settings do
   defp broadcast_env_vars_change(message) do
     Phoenix.PubSub.broadcast(Livebook.PubSub, "settings", message)
   end
+
+  @doc """
+  Sets default file system.
+  """
+  @spec set_default_file_system(file_system_id()) :: :ok
+  def set_default_file_system(file_system_id) do
+    storage().insert(:settings, "global", default_file_system_id: file_system_id)
+  end
+
+  @doc """
+  Settings default file system
+  """
+  def default_file_system do
+    {_id, file_system} =
+      Enum.find(file_systems(), fn {id, _file_system} ->
+        id == default_file_system_id
+      end)
```

Comment:
You should be able to use `storage().fetch(:filesystems, id)` or similar, no? And if it returns `:error`, you fallback to local.

Reply:
No, because storage().fetch(:filesystems, id) returns the property of the file system not the actual file system

Reply:
You can call `storage_to_fs` afterwards to get the actual fs from storage.